components/core/src/os/net/mod.rs

```diff
@@ -22,3 +22,71 @@ mod imp;
 mod imp;
 
 pub use self::imp::*;
+
+extern crate dns_lookup;
+
+use dns_lookup::{getaddrinfo,
+                 AddrInfoHints};
+use std::io;
+
+// lookup_fqdn returns a vector of fqdn that resolves the provided hostname.
+//
+// Since the underlying crate is platform agnostic, this function is as well,
+// we only implement a single function `ai_canonname()` that returns the right
+// hint flag for the running operating system.
+pub fn lookup_fqdn(hostname: &str) -> io::Result<Vec<String>> {
+    let hints = AddrInfoHints { flags: ai_canonname(),
```

Comment:
Since the `ai_cannonname()` functions are so small, and only used here, it might be a bit cleaner to model things a bit differently:

```
    #[cfg(unix)]
    let flags = libc::AI_CANONNAME;

    #[cfg(windows)]
    let flags = winapi::shared::ws2def::AI_CANONNAME;

    let hints = AddrInfoHints { flags, ..AddrInfoHints::default() };
```